File: LeetCode/Simulation/Golang/Leetcode933-最近的请求次数.go
package sbgo

type RecentCounter struct {
	q []int
}

func Constructor() RecentCounter {
	q := []int{}
	return RecentCounter{q}
}

func (this *RecentCounter) Ping(t int) int {
	this.q = append(this.q, t)
	for t-this.q[0] > 3000 {
		this.q = this.q[1:]
	}
	return len(this.q)
}

File: LeetCode/Simulation/Python/Leetcode933-最近的请求次数.py
class RecentCounter:
    def __init__(self):
        self.q = deque()

    def ping(self, t: int) -> int:
        q = self.q
        q.append(t)
        while t - q[0] > 3000:
            q.popleft()
        return len(q)
